ext/gd/gd.c

```diff
@@ -2293,6 +2293,11 @@ PHP_FUNCTION(imagegammacorrect)
 
 	gamma = input / output;
 
+	if (UNEXPECTED(!zend_finite(gamma))) {
+		zend_value_error("An input divided by an output must be finite");
+		RETURN_THROWS();
+	}
```

Comment:
Doesn't it make more sense to check that input and output are finite? This would also allow us to indicate which arg num is the problem. As I think the problem can be contained by getting rid of those values.

But I do wonder if this problem is not more widespread as I expect quite a few things could possibly break with NAN/INF being passed... maybe we should add a new ZPP specifier in master (like Z_PARAM_DOUBLE_FINITE)